client/txn.go

```diff
@@ -666,7 +666,12 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	if elideEndTxn && pErr == nil {
 		// Check that read only transactions do not violate their deadline.
 		if endTxnRequest.Deadline != nil {
-			if endTxnRequest.Deadline.Less(txn.Proto.Timestamp) {
+			t := txn.Proto.Timestamp
```

Comment:
I see. Debugged more. It seems there is a case where `NewTransaction` is not called and `Timestamp` is not initialized.

1) Request contains only an `EndTransactionRequest`.
2) `Txn.send` elides the `EndTransactionRequest.
3) `Txn.db.send` is called. `Db.send` finishes without actually sending a request with `TxnCoordinator`
4) Deadline check happens while `txn.Proto.TimeStamp` is still zero.

Will try writing a test to make sure.